### DISCUSSION TITLE
Garage Door Opener MSG100 with custom MQTT broker

### DISCUSSION THREAD

**Discussion by demichve** [krahabb/meross_lan#376] [General]
Hi,
Thanks to @krahabb for this nice work.
I am not sure, if someone can help me here, because I am not using Home Assistant but OpenHAB.
I was able to pair my MSG100 with my Mosquitto. I was also able to open and close the garage door. But this is not reliable. Sometime it works, sometimes not. When checking the  MQTT traffic, the device is permanently sending Appliance.Control.Bind messages (every 6 seconds). I guess, this could be the reason, why the device do not react on all my MQTT messages.
If I got it right, the HA integration is able to send the right responses for this type of binding process.
Perhaps I can send such MQTT messages manually to finalize this?

Best Regards,
Michael

**Comment by krahabb** [comment]
Hello @demichve.
That message is (almost not very) well known here and only recently I was able to implement a response for that.
You can find the implementation [here](https://github.com/krahabb/meross_lan/blob/21beed11e391a87d5a879b24ee60e3e837939b21/custom_components/meross_lan/__init__.py#L188) but in short you just need to reply with a (method) SETACK and an empty payload on the same namespace. This looks like quieting the device. There might be other 'session messages' though depending on the device type and in this regard, when you receive PUSHES in that phase, almost always you have to reply with another same PUSH (copying over also the `messageid` `timestamp` and `sign`, basically the full header) like [this](https://github.com/krahabb/meross_lan/blob/21beed11e391a87d5a879b24ee60e3e837939b21/custom_components/meross_lan/__init__.py#L230C15-L230C58)

You can find more info about the MQTT session establishment in this [issue](https://github.com/krahabb/meross_lan/issues/346)

**Comment by demichve** [comment]
@krahabb 
You are my hero!
It works like a charm. Just the right MQTT message was sufficient to make the device quiet. And now it works reliable.
Thank you very much!
